# 异常捕捉

```python
try:
    # 检测异常的代码
    pass
except ValueError:
    print("数值错误")
except TypeError:
    print("类型错误")
except Exception as result:
    print("未知错误 %s " % result)
else:
    # 没有异常才会被执行的代码
    pass
finally:
    # 无论是否发生异常都会被执行的代码
    pass
```

**如果函数或方法发生异常时，程序并不会立即终止，而是会将异常传递给调用函数或方法的一方，如果在主程序中异常还是没得到处理，程序就会终止**

* Exception 是python中专门针对异常的类

## 主动抛出异常

* 创建Exception对象